integration_tests/src/main/python/map_test.py

```diff
@@ -91,6 +91,48 @@ def test_map_entries(data_gen):
                 'map_entries(a)'))
 
 
+@pytest.mark.parametrize('data_gen', supported_key_map_gens, ids=idfn)
+def test_map_from_entries(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark: unary_op_df(spark, data_gen).selectExpr(
+                'map_from_entries(map_entries(a))'),
+            conf={'spark.sql.mapKeyDedupPolicy': 'LAST_WIN'})
+
+
+@pytest.mark.parametrize('key_gen,value_gen', [
+    (IntegerGen(nullable=False), IntegerGen()),
+    (StringGen(nullable=False), StringGen()),
+    (LongGen(nullable=False), DoubleGen()),
+    (IntegerGen(nullable=False), ArrayGen(IntegerGen())),
+    (StringGen(nullable=False), StructGen([['child', IntegerGen()]])),
+], ids=idfn)
+def test_map_from_entries_basic(key_gen, value_gen):
+    # Create an array of struct<key, value> and convert to map
+    struct_gen = StructGen([
+        ('key', key_gen),
+        ('value', value_gen)
+    ], nullable=False)
+    array_gen = ArrayGen(struct_gen, max_length=10)
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark: unary_op_df(spark, array_gen).selectExpr(
+                'map_from_entries(a)'),
+            conf={'spark.sql.mapKeyDedupPolicy': 'LAST_WIN'})
+
+
+def test_map_from_entries_dup_exception():
+    # Test that duplicate keys throw an exception with EXCEPTION policy
+    struct_gen = StructGen([
+        ('key', IntegerGen(nullable=False)),
```

Comment:
Isn't there the possibility that this will NOT generate duplicate keys?  Should we make sure that it does with a RepeatSeqGen that has fewer entries it repeats than there are min rows in the array.

Reply:
yes there is, `RepeatSeqGen` works, will push